plotpal/volumes.py

```diff
@@ -0,0 +1,373 @@
+            ax.plot_wireframe(x, y, z, ccount=1, rcount=1, linewidth=1, color='black')
+        
+        ax.view_init(self.azim, self.elev)
+        cb = self._setup_colorbar(cmap, cax, vmin, vmax)
+        self.first = False
+        return surf, cb
+
+
+class BoxPlotter(SingleTypeReader):
+    """
+    A class for plotting 3D boxes of dedalus data.
+
+    # Public Methods
+    - __init__()
+    - setup_grid()
+    - add_colormesh()
+    - plot_colormeshes()
+
+    # Attributes
+        colormeshes (list) :
+            A list of Colormesh objects
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initializes the slice plotter.
+
+        # Arguments
+            *args, **kwargs : Additional keyword arguments for super().__init__() 
+        """
+        self.grid = None
+        self.cutout = False #this line says syntax error
+        super(BoxPlotter, self).__init__(*args, distribution='even-write', **kwargs)
+        self.counter = 0
+        self.boxes = []
+        
+
+    def setup_grid(self, *args, **kwargs):
+        """ Initialize the plot grid for the colormeshes """
+        self.grid = RegularColorbarPlotGrid(*args, **kwargs, threeD=True)
+
+
+    def add_box(self, *args, **kwargs):
+        self.boxes.append((self.counter, Box(*args, **kwargs)))
+        self.counter += 1
+    
+    def add_cutout_box(self, *args, **kwargs):
+        self.boxes.append((self.counter, Box(*args, **kwargs))
+        self.cutout = True #need an argument for if statement on line 382
+        self.counter += 1
+
+    def _groom_grid(self):
+        """ Assign colormeshes to axes subplots in the plot grid """
+        axs, caxs = [], []
+        for nr in range(self.grid.nrows):
+            for nc in range(self.grid.ncols):
+                k = 'ax_{}-{}'.format(nr, nc)
+                if k in self.grid.axes.keys():
+                    axs.append(self.grid.axes[k])
+                    caxs.append(self.grid.cbar_axes[k])
+        return axs, caxs
+    
+    
+    
+   
+    def plot_boxes(self, start_fig=1, dpi=200, **kwargs):
+        """
+        Plot figures of the 2D dedalus data slices at each timestep.
+
+        # Arguments
+            start_fig (int) :
+                The number in the filename for the first write.
+            dpi (int) :
+                The pixel density of the output image
+            kwargs :
+                extra keyword args for matplotlib.pyplot.pcolormesh
+        """
+        
+        with self.my_sync:
+            axs, caxs = self._groom_grid()
+            tasks = []
+            for k, bx in self.boxes:
+                if bx.left not in tasks:
+                    tasks.append(bx.left)
+                if bx.right not in tasks:
+                    tasks.append(bx.right)
+                if bx.top not in tasks:
+                    tasks.append(bx.top)
+                if self.cutout is True:
```

Comment:
so that I can add these so that you can plot a cut out box and a normal box if needed